PrismOutlook/Core/Dialogs/RegionDialogService.cs

```diff
@@ -29,58 +32,89 @@ public void Show(string name, IDialogParameters dialogParameters, Action<IDialog
 
             IRegion region = scopedRegionManager.Regions[RegionNames.ContentRegion];
 
-            region.RequestNavigate(name);
-
-            var activeView = region.ActiveViews.FirstOrDefault() as FrameworkElement;
-            IDialogAware dialogAware = activeView.DataContext as IDialogAware;
-            if (dialogAware == null)
-                throw new ArgumentNullException("dialogAware");
-
-            dialogAware.OnDialogOpened(dialogParameters);
-
             Action<IDialogResult> requestCloseHandler = null;
+
             requestCloseHandler = (result) =>
-            {
-                window.Result = result;
-                window.Close();
-            };
+                       {
+                           window.Result = result;
+                           window.Close();
+                       };
 
             CancelEventHandler closingHandler = null;
             closingHandler = (o, e) =>
             {
-                if (!dialogAware.CanCloseDialog())
-                    e.Cancel = true;
+                if (o is ContentControl co && co.Content is FrameworkElement fe && fe.DataContext is IDialogAware dialogAware)
```

Comment:
did you test this? Looking at this 'o' will be the Window. Also, how can you guarantee that the Content's DataContext will be set, let alone the correct DataConext to check? Keep in mind that a dialog will most likely have many regions.

Reply:
I have a gut feeling that we are mixing issues here. Correct me if I am wrong. Since we are using region navigation data context will be set by prism itself unlike the current prism dialog service where you have to manually set the content and data context. Am I missing something here ?

Reply:
Oh yeah now I think I get your point. In my Dialog window I have set the region name to the windows content itself. So my navigation is replacing the entire content with the navigated view and the view model is set properly by the navigation.

Reply:
This is my RegionDialogWindow

```csharp
<dx:ThemedWindow
    x:Class="Jasmine.Desktop.Prism.RegionDialogWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:dx="http://schemas.devexpress.com/winfx/2008/xaml/core"
    xmlns:core="clr-namespace:Jasmine.Core;assembly=Jasmine.Core"
    Title="{Binding Title}" 
    xmlns:prism="http://prismlibrary.com/" prism:RegionManager.RegionName="{x:Static core:KnownRegions.ContentRegion}"
    WindowStartupLocation="CenterOwner">

</dx:ThemedWindow>
```

Reply:
Right, in a multi-region scenario it's not so simple :)